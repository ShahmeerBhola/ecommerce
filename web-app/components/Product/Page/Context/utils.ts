import isEmpty from 'lodash/isEmpty';

import type { TProductContext } from './types';

export const getSingleAttributes = (_attributes?: StringMap<string[]>): StringMap => {
    let singleAttributes = {};

    for (const [key, value] of Object.entries(_attributes || {})) {
        if (value.length === 1) {
            singleAttributes = {
                ...singleAttributes,
                [key]: value[0],
            };
        }
    }

    return singleAttributes;
};

export const getAvailableAttributes = (
    variations: WordPress.ProductVariation[],
    newAttributes: StringMap,
): StringMap<string[]> => {
    const availableAttribute: StringMap<string[]> = {};

    for (const { attributes } of variations) {
        let isAvailableAttribute = true;

        for (const [key, value] of Object.entries(newAttributes)) {
            isAvailableAttribute = isAvailableAttribute && attributes[key] === value;
        }

        if (isAvailableAttribute) {
            for (const [key, value] of Object.entries(attributes)) {
                const currentAttribute = availableAttribute[key] || [];

                if (currentAttribute.indexOf(value) > -1) {
                    continue;
                }

                availableAttribute[key] = [...currentAttribute, value];
            }
        }
    }

    return availableAttribute;
};

const getAttributes = (
    variations: WordPress.ProductVariation[],
    selectedAttributes: {} | StringMap,
): TProductContext.CheckAttributesType => {
    const availableAttributes = getAvailableAttributes(variations, selectedAttributes);
    const singleAttributes = getSingleAttributes(availableAttributes);

    return {
        currentAttributes: {
            ...selectedAttributes,
            ...singleAttributes,
        },
        availableAttributes,
    };
};

export const checkAttributes = (state: TProductContext.State, payload: any): TProductContext.CheckAttributesType => {
    const { currentAttributes, product } = state;
    const { variations } = product as WordPress.Product;
    const newAttributes = {
        ...currentAttributes,
        ...payload,
    };

    const attributes = getAttributes(variations, newAttributes);

    if (isEmpty(attributes.availableAttributes)) {
        const resetAttributes = {
            ...payload,
        };

        return getAttributes(variations, resetAttributes);
    }

    return attributes;
};

export const isActiveAttribute = (attributes: StringMap, currentAttributes: StringMap): boolean => {
    return Object.entries(attributes).reduce<boolean>((isSelectedAttribute, attribute) => {
        const [key, value] = attribute;
        const matchedAttribute = value === '' || currentAttributes[key] === value;

        return isSelectedAttribute && matchedAttribute;
    }, true);
};
