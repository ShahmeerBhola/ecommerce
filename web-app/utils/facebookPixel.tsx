import React from 'react';
import config from '../config';

const { facebookPixelID } = config;

// https://www.facebook.com/business/help/402791146561655?id=1205376682832142
const event = <P,>(eventName: string, params?: P): void => {
  if (facebookPixelID !== null) {
    if (params !== undefined) {
      window.fbq('track', eventName, params);
    } else {
      window.fbq('track', eventName);
    }
  }
};

export const addLineItemToCart = ({ sku, name, price }: WordPress.ProductBase, type = 'line_item'): void =>
  event<facebook.Pixel.AddToCartParameters>('AddToCart', {
    content_name: name,
    content_type: type,
    content_ids: [sku],
    value: price,
    currency: config.defaultCurrencyCode,
  });

export const addWheelTirePackageToCart = ({ wheel, tire, addOns }: Cart.WheelTirePackage): void => {
  const addWheelTirePackageItemToCart = (product: WordPress.ProductBase): void =>
    addLineItemToCart(product, 'wheel_tire_package');

  addWheelTirePackageItemToCart(wheel);
  addWheelTirePackageItemToCart(tire);
  addOns.map((addOn) => addWheelTirePackageItemToCart(addOn));
};

export const initiateCheckout = (): void => event('InitiateCheckout');
export const completePurchase = (params: facebook.Pixel.PurchaseParameters): void =>
  event<facebook.Pixel.PurchaseParameters>('Purchase', params);

export const FacebookPixelScripts: React.FunctionComponent = () => {
  {
    if (facebookPixelID !== null) {
      return (
        <>
          <script
            dangerouslySetInnerHTML={{
              __html: `
            !function(f,b,e,v,n,t,s)
{if(f.fbq)return;n=f.fbq=function(){n.callMethod ?
              n.callMethod.apply(n, arguments) : n.queue.push(arguments)};
if(!f._fbq)f._fbq=n;n.push=n;n.loaded=!0;n.version='2.0';
n.queue=[];t=b.createElement(e);t.async=!0;
t.src=v;s=b.getElementsByTagName(e)[0];
s.parentNode.insertBefore(t,s)}(window,document,'script',
'https://connect.facebook.net/en_US/fbevents.js');
 fbq('init', '${facebookPixelID}');
fbq('track', 'PageView');
          `,
            }}
          />
          <noscript>
            <img
              height="1"
              width="1"
              src={`https://www.facebook.com/tr?id=${facebookPixelID}&ev=PageView
&noscript=1`}
            />
          </noscript>
        </>
      );
    }
  }
  return null;
};
