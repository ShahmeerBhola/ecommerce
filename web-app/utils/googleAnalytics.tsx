import React from 'react';
import config from '../config';
import { isBrowserSide } from './';

const { googleAnalyticsTrackingID, googleAdsConversionID } = config;

// https://developers.google.com/analytics/devguides/collection/gtagjs/pages
// https://github.com/vercel/next.js/tree/canary/examples/with-google-analytics
export const pageview = (url: string): void => {
  if (googleAnalyticsTrackingID !== null && isBrowserSide()) {
    window.gtag('config', googleAnalyticsTrackingID, {
      page_path: url,
    });
  }
};

// https://developers.google.com/analytics/devguides/collection/gtagjs/events
const event = (action: string, eventParams: any): void => {
  if (googleAnalyticsTrackingID !== null && isBrowserSide()) {
    window.gtag('event', action, eventParams);
  }
};

const conversionEvent = (sendTo: string, eventParams?: StringMap): void =>
  event('conversion', { send_to: sendTo, ...eventParams });

export const sendOrderConfirmationConversionEvent = (orderId: string): void =>
  conversionEvent('AW-479563831/zZZcCPqr3-gBELeg1uQB', { transaction_id: orderId });

export const sendContactFormConversionEvent = (): void => conversionEvent('AW-479563831/6i-DCKyL6-gBELeg1uQB');

export const sendRequestForQuoteEvent = (): void => conversionEvent('AW-479563831/C9PsCJv7_-gBELeg1uQB');

// https://developers.google.com/analytics/devguides/collection/gtagjs/enhanced-ecommerce
export const sendEnhancedEcommerceEvent = (
  eee: GoogleAnalytics.EnhancedEcommerceEvent,
  items: GoogleAnalytics.Product[],
): void => event(eee, { items });

export const sendEnhancedEcommercePurchaseEvent = (data: GoogleAnalytics.PurchaseEvent): void =>
  event('purchase', data);

export const sendEnhancedEcommerceCheckoutOptionEvent = (data: GoogleAnalytics.CheckoutProgressEvent): void =>
  event('set_checkout_option', data);

export const enhancedEcommerceLineItemListName = 'Line Items';
export const enhancedEcommerceWheelTirePackageListName = 'Wheel Tire Package';
export const enhancedEcommerceWheelTirePackageAddOnsListName = 'Wheel Tire Package Add Ons';

export const GoogleAnalyticsAndAdsScripts: React.FunctionComponent = () => {
  {
    if (googleAnalyticsTrackingID !== null && googleAdsConversionID !== null) {
      return (
        <>
          <script async src={`https://www.googletagmanager.com/gtag/js?id=${googleAnalyticsTrackingID}`} />
          <script
            dangerouslySetInnerHTML={{
              __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '${googleAnalyticsTrackingID}', {
              page_path: window.location.pathname,
            });
            gtag('config', '${googleAdsConversionID}')
          `,
            }}
          />
        </>
      );
    }
  }
  return null;
};
