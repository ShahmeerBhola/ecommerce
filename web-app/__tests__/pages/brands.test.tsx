import React from 'react';
import renderer from 'react-test-renderer';
import BrandsPage from '../../pages/brands';
import { ToastProvider, generateMockBrand } from '../utils';

describe('BrandsPage component', () => {
  test('it renders properly', () => {
    const tree = renderer
      .create(
        <ToastProvider>
          <BrandsPage brands={[generateMockBrand(1), generateMockBrand(2), generateMockBrand(3)]} />
        </ToastProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
