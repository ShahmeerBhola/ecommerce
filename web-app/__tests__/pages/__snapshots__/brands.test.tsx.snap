// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandsPage component it renders properly 1`] = `
Array [
  <header
    className="fixed z-10 w-full py-5 text-white bg-black"
  >
    <div
      className="container mx-auto px-5   flex items-center justify-between"
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <img
          alt="Standout Specialties"
          className="mx-auto"
          height={79}
          src="https://storage.googleapis.com/standout-specialties-public-assets/logo.png"
          width={161}
        />
      </a>
      <nav
        className="hidden sm:flex items-center"
      >
        <a
          className="mr-10"
          href="/store"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Store
        </a>
        <a
          className="mr-10"
          href="/brands"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Brands
        </a>
        <a
          className="mr-10"
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Contact
        </a>
        <a
          className="mr-10"
          href="/cart"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="relative"
          >
            <svg
              fill="none"
              height="34"
              viewBox="0 0 38 34"
              width="34"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.01167 26.2134H23.2637L24.2063 22.5357H4.2741L0 5.86091H28.4804L29.9827 0H38V2.18315H31.7304L25.5486 26.3006C27.3481 26.6775 28.7015 28.2406 28.7015 30.1067C28.7015 32.2535 26.9108 34 24.7097 34C22.5086 34 20.718 32.2534 20.718 30.1067C20.718 29.4935 20.8646 28.9131 21.1247 28.3966H11.5967C11.8569 28.9131 12.0034 29.4935 12.0034 30.1067C12.0034 32.2535 10.2127 34 8.01167 34C5.81065 34 4.0199 32.2534 4.0199 30.1067C4.01997 27.96 5.81072 26.2134 8.01167 26.2134Z"
                fill="white"
              />
            </svg>
          </div>
        </a>
      </nav>
      <button
        className="block sm:hidden flex items-center px-3 py-2 border rounded text-red-100 border-red-100 hover:text-red-200 hover:border-red-200 focus:outline-none"
        onClick={[Function]}
      >
        <svg
          className="fill-current h-3 w-3"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Menu
          </title>
          <path
            d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
          />
        </svg>
      </button>
    </div>
    <nav
      className="hidden sm:hidden mt-2 py-2 text-center bg-gray-600"
    >
      <a
        className="block py-2 hover:bg-red-100 hover:bg-opacity-25"
        href="/store"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Store
      </a>
      <a
        className="block py-2 hover:bg-red-100 hover:bg-opacity-25"
        href="/brands"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Brands
      </a>
      <a
        className="block py-2 hover:bg-red-100 hover:bg-opacity-25"
        href="/contact"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Contact
      </a>
      <a
        className="block py-2 hover:bg-red-100 hover:bg-opacity-25"
        href="/cart"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Cart
      </a>
    </nav>
  </header>,
  <div
    className="container mx-auto px-5 md:py-40 py-5 min-h-80"
  >
    <h1
      className="font-primary font-bold text-black text-3xl uppercase"
    >
      Brands
    </h1>
    <div
      className="flex flex-wrap"
    >
      <div
        className="h-cube p-2 box-border w-full md:w-1/2 lg:w-1/3"
      >
        <a
          href="/brand/test-brand-1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="Test Brand"
            src="https://images.customwheeloffset.com/brand-logos/Fuel.png"
          />
        </a>
      </div>
      <div
        className="h-cube p-2 box-border w-full md:w-1/2 lg:w-1/3"
      >
        <a
          href="/brand/test-brand-2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="Test Brand"
            src="https://images.customwheeloffset.com/brand-logos/Fuel.png"
          />
        </a>
      </div>
      <div
        className="h-cube p-2 box-border w-full md:w-1/2 lg:w-1/3"
      >
        <a
          href="/brand/test-brand-3"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="Test Brand"
            src="https://images.customwheeloffset.com/brand-logos/Fuel.png"
          />
        </a>
      </div>
    </div>
  </div>,
  <div
    className="bg-gray-600"
  >
    <div
      className="container mx-auto px-5 md:py-20 py-5 "
    >
      <footer
        className="flex flex-wrap justify-between"
      >
        <div
          className="w-full md:w-1/5 text-center"
        >
          <img
            alt="Standout Specialties"
            className="mx-auto"
            height={79}
            src="https://storage.googleapis.com/standout-specialties-public-assets/logo.png"
            width={161}
          />
          <div
            className="mt-10"
          >
            <a
              aria-label="facebook"
              className="social-icon mx-2 h-40px w-40px"
              href="https://www.facebook.com/standoutspecialties"
              rel="noopener noreferrer"
              style={
                Object {
                  "display": "inline-block",
                  "height": "50px",
                  "overflow": "hidden",
                  "position": "relative",
                  "verticalAlign": "middle",
                  "width": "50px",
                }
              }
              target="_blank"
            >
              <div
                className="social-container"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
                }
              >
                <svg
                  className="social-svg"
                  style={
                    Object {
                      "borderRadius": "50%",
                      "fillRule": "evenodd",
                      "height": "100%",
                      "left": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                  viewBox="0 0 64 64"
                >
                  <g
                    className="social-svg-background"
                    style={
                      Object {
                        "MozTransition": "fill 170ms ease-in-out",
                        "OTransition": "fill 170ms ease-in-out",
                        "WebkitTransition": "fill 170ms ease-in-out",
                        "fill": "transparent",
                        "msTransition": "fill 170ms ease-in-out",
                        "transition": "fill 170ms ease-in-out",
                      }
                    }
                  >
                    <circle
                      cx="32"
                      cy="32"
                      r="31"
                    />
                  </g>
                  <g
                    className="social-svg-icon"
                    style={
                      Object {
                        "MozTransition": "fill 170ms ease-in-out",
                        "OTransition": "fill 170ms ease-in-out",
                        "WebkitTransition": "fill 170ms ease-in-out",
                        "fill": "transparent",
                        "msTransition": "fill 170ms ease-in-out",
                        "transition": "fill 170ms ease-in-out",
                      }
                    }
                  >
                    <path
                      d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
                    />
                  </g>
                  <g
                    className="social-svg-mask"
                    style={
                      Object {
                        "MozTransition": "fill 170ms ease-in-out",
                        "OTransition": "fill 170ms ease-in-out",
                        "WebkitTransition": "fill 170ms ease-in-out",
                        "fill": "#FFFFFF",
                        "msTransition": "fill 170ms ease-in-out",
                        "transition": "fill 170ms ease-in-out",
                      }
                    }
                  >
                    <path
                      d="M0,0v64h64V0H0z M39.6,22l-2.8,0c-2.2,0-2.6,1.1-2.6,2.6V28h5.3l-0.7,5.3h-4.6V47h-5.5V33.3H24V28h4.6V24 c0-4.6,2.8-7,6.9-7c2,0,3.6,0.1,4.1,0.2V22z"
                    />
                  </g>
                </svg>
              </div>
            </a>
            <a
              aria-label="instagram"
              className="social-icon mx-2 h-40px w-40px"
              href="https://www.instagram.com/standoutspecialties"
              rel="noopener noreferrer"
              style={
                Object {
                  "display": "inline-block",
                  "height": "50px",
                  "overflow": "hidden",
                  "position": "relative",
                  "verticalAlign": "middle",
                  "width": "50px",
                }
              }
              target="_blank"
            >
              <div
                className="social-container"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
                }
              >
                <svg
                  className="social-svg"
                  style={
                    Object {
                      "borderRadius": "50%",
                      "fillRule": "evenodd",
                      "height": "100%",
                      "left": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                  viewBox="0 0 64 64"
                >
                  <g
                    className="social-svg-background"
                    style={
                      Object {
                        "MozTransition": "fill 170ms ease-in-out",
                        "OTransition": "fill 170ms ease-in-out",
                        "WebkitTransition": "fill 170ms ease-in-out",
                        "fill": "transparent",
                        "msTransition": "fill 170ms ease-in-out",
                        "transition": "fill 170ms ease-in-out",
                      }
                    }
                  >
                    <circle
                      cx="32"
                      cy="32"
                      r="31"
                    />
                  </g>
                  <g
                    className="social-svg-icon"
                    style={
                      Object {
                        "MozTransition": "fill 170ms ease-in-out",
                        "OTransition": "fill 170ms ease-in-out",
                        "WebkitTransition": "fill 170ms ease-in-out",
                        "fill": "transparent",
                        "msTransition": "fill 170ms ease-in-out",
                        "transition": "fill 170ms ease-in-out",
                      }
                    }
                  >
                    <path
                      d="M43.5,29.7h-2.6c0.2,0.7,0.3,1.5,0.3,2.3 c0,5.1-4.1,9.2-9.2,9.2c-5.1,0-9.2-4.1-9.2-9.2c0-0.8,0.1-1.6,0.3-2.3h-2.6v12.7c0,0.6,0.5,1.2,1.2,1.2h20.8c0.6,0,1.2-0.5,1.2-1.2 V29.7z M43.5,21.6c0-0.6-0.5-1.2-1.2-1.2h-3.5c-0.6,0-1.2,0.5-1.2,1.2v3.5c0,0.6,0.5,1.2,1.2,1.2h3.5c0.6,0,1.2-0.5,1.2-1.2V21.6z  M32,26.2c-3.2,0-5.8,2.6-5.8,5.8c0,3.2,2.6,5.8,5.8,5.8s5.8-2.6,5.8-5.8C37.8,28.8,35.2,26.2,32,26.2 M43.5,47H20.5 c-1.9,0-3.5-1.6-3.5-3.5V20.5c0-1.9,1.5-3.5,3.5-3.5h23.1c1.9,0,3.5,1.5,3.5,3.5v23.1C47,45.4,45.5,47,43.5,47"
                    />
                  </g>
                  <g
                    className="social-svg-mask"
                    style={
                      Object {
                        "MozTransition": "fill 170ms ease-in-out",
                        "OTransition": "fill 170ms ease-in-out",
                        "WebkitTransition": "fill 170ms ease-in-out",
                        "fill": "#FFFFFF",
                        "msTransition": "fill 170ms ease-in-out",
                        "transition": "fill 170ms ease-in-out",
                      }
                    }
                  >
                    <path
                      d="M41.2,32c0,5.1-4.1,9.2-9.2,9.2c-5.1,0-9.2-4.1-9.2-9.2c0-0.8,0.1-1.6,0.3-2.3h-2.6v12.7c0,0.6,0.5,1.2,1.2,1.2 h20.8c0.6,0,1.2-0.5,1.2-1.2V29.7h-2.6C41.1,30.4,41.2,31.2,41.2,32z M32,37.8c3.2,0,5.8-2.6,5.8-5.8c0-3.2-2.6-5.8-5.8-5.8 c-3.2,0-5.8,2.6-5.8,5.8C26.2,35.2,28.8,37.8,32,37.8z M42.4,20.5h-3.5c-0.6,0-1.2,0.5-1.2,1.2v3.5c0,0.6,0.5,1.2,1.2,1.2h3.5 c0.6,0,1.2-0.5,1.2-1.2v-3.5C43.5,21,43,20.5,42.4,20.5z M0,0v64h64V0H0z M47,43.5c0,1.9-1.5,3.5-3.5,3.5H20.5 c-1.9,0-3.5-1.6-3.5-3.5V20.5c0-1.9,1.5-3.5,3.5-3.5h23.1c1.9,0,3.5,1.5,3.5,3.5V43.5z"
                    />
                  </g>
                </svg>
              </div>
            </a>
            <a
              aria-label="youtube"
              className="social-icon mx-2 h-40px w-40px"
              href="https://www.youtube.com/channel/UC_L8_2r214VcNWAHeV8nMXw"
              rel="noopener noreferrer"
              style={
                Object {
                  "display": "inline-block",
                  "height": "50px",
                  "overflow": "hidden",
                  "position": "relative",
                  "verticalAlign": "middle",
                  "width": "50px",
                }
              }
              target="_blank"
            >
              <div
                className="social-container"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
                }
              >
                <svg
                  className="social-svg"
                  style={
                    Object {
                      "borderRadius": "50%",
                      "fillRule": "evenodd",
                      "height": "100%",
                      "left": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                  viewBox="0 0 64 64"
                >
                  <g
                    className="social-svg-background"
                    style={
                      Object {
                        "MozTransition": "fill 170ms ease-in-out",
                        "OTransition": "fill 170ms ease-in-out",
                        "WebkitTransition": "fill 170ms ease-in-out",
                        "fill": "transparent",
                        "msTransition": "fill 170ms ease-in-out",
                        "transition": "fill 170ms ease-in-out",
                      }
                    }
                  >
                    <circle
                      cx="32"
                      cy="32"
                      r="31"
                    />
                  </g>
                  <g
                    className="social-svg-icon"
                    style={
                      Object {
                        "MozTransition": "fill 170ms ease-in-out",
                        "OTransition": "fill 170ms ease-in-out",
                        "WebkitTransition": "fill 170ms ease-in-out",
                        "fill": "transparent",
                        "msTransition": "fill 170ms ease-in-out",
                        "transition": "fill 170ms ease-in-out",
                      }
                    }
                  >
                    <path
                      d="M46.7,26c0,0-0.3-2.1-1.2-3c-1.1-1.2-2.4-1.2-3-1.3C38.3,21.4,32,21.4,32,21.4h0 c0,0-6.3,0-10.5,0.3c-0.6,0.1-1.9,0.1-3,1.3c-0.9,0.9-1.2,3-1.2,3S17,28.4,17,30.9v2.3c0,2.4,0.3,4.9,0.3,4.9s0.3,2.1,1.2,3 c1.1,1.2,2.6,1.2,3.3,1.3c2.4,0.2,10.2,0.3,10.2,0.3s6.3,0,10.5-0.3c0.6-0.1,1.9-0.1,3-1.3c0.9-0.9,1.2-3,1.2-3s0.3-2.4,0.3-4.9 v-2.3C47,28.4,46.7,26,46.7,26z M28.9,35.9l0-8.4l8.1,4.2L28.9,35.9z"
                    />
                  </g>
                  <g
                    className="social-svg-mask"
                    style={
                      Object {
                        "MozTransition": "fill 170ms ease-in-out",
                        "OTransition": "fill 170ms ease-in-out",
                        "WebkitTransition": "fill 170ms ease-in-out",
                        "fill": "#FFFFFF",
                        "msTransition": "fill 170ms ease-in-out",
                        "transition": "fill 170ms ease-in-out",
                      }
                    }
                  >
                    <path
                      d="M0,0v64h64V0H0z M47,33.1c0,2.4-0.3,4.9-0.3,4.9s-0.3,2.1-1.2,3c-1.1,1.2-2.4,1.2-3,1.3 C38.3,42.5,32,42.6,32,42.6s-7.8-0.1-10.2-0.3c-0.7-0.1-2.2-0.1-3.3-1.3c-0.9-0.9-1.2-3-1.2-3S17,35.6,17,33.1v-2.3 c0-2.4,0.3-4.9,0.3-4.9s0.3-2.1,1.2-3c1.1-1.2,2.4-1.2,3-1.3c4.2-0.3,10.5-0.3,10.5-0.3h0c0,0,6.3,0,10.5,0.3c0.6,0.1,1.9,0.1,3,1.3 c0.9,0.9,1.2,3,1.2,3s0.3,2.4,0.3,4.9V33.1z M28.9,35.9l8.1-4.2l-8.1-4.2L28.9,35.9z"
                    />
                  </g>
                </svg>
              </div>
            </a>
          </div>
        </div>
        <div
          className="md:flex flex-col hidden"
        >
          <p
            className="font-secondary font-normal text-red-100 text-sm font-extrabold pb-6"
          >
            Our Site
          </p>
          <a
            href="/store"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <p
              className="font-secondary font-normal text-white text-sm pt-5"
            >
              Store
            </p>
          </a>
          <a
            href="/brands"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <p
              className="font-secondary font-normal text-white text-sm pt-5"
            >
              Brands
            </p>
          </a>
        </div>
        <div
          className="md:flex flex-col hidden"
        >
          <p
            className="font-secondary font-normal text-red-100 text-sm font-extrabold pb-6"
          >
            Our Site
          </p>
          <a
            href="/contact"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <p
              className="font-secondary font-normal text-white text-sm pt-5"
            >
              Contact
            </p>
          </a>
          <a
            href="/request_quote"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <p
              className="font-secondary font-normal text-white text-sm pt-5"
            >
              Request a Quote
            </p>
          </a>
        </div>
        <div
          className="flex flex-col"
        >
          <p
            className="font-secondary font-normal text-red-100 text-sm font-extrabold pb-6"
          >
            Working Hours
          </p>
          <p
            className="font-secondary font-normal text-gray-200 text-sm uppercase font-bold pt-5"
          >
            Monday - Friday
              
            <span
              className="text-white"
            >
              9:30-6
            </span>
          </p>
          <p
            className="font-secondary font-normal text-gray-200 text-sm uppercase font-bold pt-5"
          >
            Saturday
              
            <span
              className="text-white"
            >
              9:30-2
            </span>
          </p>
          <hr
            className="my-5 w-full mx-auto border-2 border-solid border-red-100"
          />
          <p
            className="font-secondary font-normal text-gray-200 text-sm"
          >
            4011 Oregon Pike
            <br />
            Ephrata
            , 
            PA
             
            17522
          </p>
        </div>
        <div
          className="flex flex-col"
        >
          <p
            className="font-secondary font-normal text-red-100 text-sm font-extrabold pb-6"
          >
            Subscribe for News and Upcoming Events
          </p>
          <form
            className="font-secondary"
            onSubmit={[Function]}
          >
            <div
              className="flex flex-wrap my-3"
            >
              <div
                className="pt-2 px-2 box-border pb-5 w-full"
              >
                <input
                  className="text-sm text-gray-200 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid border-gray-100"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="pt-2 px-2 box-border pb-5 w-full"
              >
                <button
                  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none h-full w-full text-sm text-white bg-red-100 hover:bg-red-200"
                  disabled={false}
                  type="submit"
                >
                  Subscribe
                </button>
              </div>
            </div>
          </form>
        </div>
      </footer>
    </div>
  </div>,
  <div
    className="react-toast-notifications__container css-mrh2d-ToastContainer"
  />,
]
`;
